daemon/algod/api/server/common/handlers.go

```diff
@@ -89,6 +92,63 @@ func HealthCheck(ctx lib.ReqContext, context echo.Context) {
 	json.NewEncoder(w).Encode(nil)
 }
 
+// Ready is a httpHandler for route GET /ready
+// it serves "readiness" probe on if the node is healthy and fully caught-up.
+func Ready(ctx lib.ReqContext, context echo.Context) {
+	// swagger:operation GET /ready Ready
+	//---
+	//     Summary: Returns OK if healthy and fully caught up.
+	//     Produces:
+	//     - application/json
+	//     Schemes:
+	//     - http
+	//     Responses:
+	//       200:
+	//         description: OK.
+	//       500:
+	//         description: Internal Error.
+	//       503:
+	//         description: Node not ready yet.
+	//       default: { description: Unknown Error }
+	w := context.Response().Writer
+	w.Header().Set("Content-Type", "application/json")
+
+	stat, err := ctx.Node.Status()
+	code := http.StatusOK
+
+	// isReadyFromStat checks the `Node.Status()` result
+	// and decide if the node is at the latest round
+	// must satisfy following sub conditions:
+	// 1. the node is not in a fast-catchup stage
+	// 2. the node's time since last round should be [0, deadline),
+	//    while deadline = bigLambda + smallLambda = 17s
+	// 3. the node's catchup time is 0
+	isReadyFromStat := func(status node.StatusReport) bool {
+		timeSinceLastRound := status.TimeSinceLastRound().Milliseconds()
+
+		return len(status.Catchpoint) == 0 &&
+			timeSinceLastRound >= 0 &&
+			timeSinceLastRound < agreement.DeadlineTimeout().Milliseconds() &&
+			status.CatchupTime.Milliseconds() == 0
+	}
+
+	if err != nil {
+		code = http.StatusInternalServerError
+	} else if stat.StoppedAtUnsupportedRound {
+		code = http.StatusInternalServerError
+		err = fmt.Errorf("stopped at an unsupported round")
+	} else if !isReadyFromStat(stat) {
+		code = http.StatusServiceUnavailable
+		err = fmt.Errorf("ready failed as the node is catching up")
+	}
+
+	if err != nil {
+		ctx.Log.Info(err)
```

Comment:
Re logging level, I would think the first conditions above should be at `ERROR` level, but the `!isReadyFromStat()` should be `INFO`.

Reply:
see [reset log level of endpoint resp, maybe need another round of reapprove](https://github.com/algorand/go-algorand/pull/4844/commits/3d949d38e19a9469c66217496e913b781b42d55d)